### PR TITLE
add equality methods for AbstractArray and check element-wise equality on FixedArrays

### LINKED ISSUE #5: equality function with standard arrays is missing
``` julia
julia> FixedSizeArrays.Vec3(1,2,3) == [1,2,3]
false
```

whereas

``` julia
julia> ImmutableArrays.Vector2(1,2) == [1,2]
true
```

### PR DESCRIPTION
**Pull request by sjkelly**
Closes #5 and #7

### CODE REVIEW

File: src/ops.jl [hunk 1]
```diff
@@ -182,3 +182,28 @@ end
 function (*){FSV <: FixedVector}(a::FSV, b::FSV)
     FixedMatrix{eltype(FSV), 1, 1}(dot(a,b))
 end
+
+function (==){T1,T2,C,N}(a::FixedArray{T1,C,N}, b::FixedArray{T2,C,N})
+    l = prod(N.parameters)
```
Comment (SimonDanisch) [inline src/ops.jl]: Why don't you use length here?

Comment (sjkelly) [inline src/ops.jl]: No reason, but `length` is more understandable. Either way, it should become a constant.

Comment (SimonDanisch) [inline src/ops.jl]: yeah, and it keeps the "ugliness" of prod(N.parameters) at one place ;)

**Comment by SimonDanisch** [comment]
Great, thanks a lot  =)

**Pull request merged**